is/tuple_of.ts

```diff
@@ -93,6 +125,23 @@ export function isTupleOf<
       predTup,
       predRest,
     );
+  } else {
+    return rewriteName(
+      (x: unknown): x is [...TupleOf<T>, ...PredicateType<R>] => {
+        if (!isArray(x) || x.length < (predTup.length + predTrail.length)) {
+          return false;
+        }
+        const rest = x.slice(predTup.length, -predTrail.length);
+        const trail = x.slice(-predTrail.length);
+        return predTup.every((pred, i) => pred(x[i])) &&
+          predTrail.every((pred, i) => pred(trail[i])) &&
```

Comment:
I think we can calculate `offset` so that we can avoid copying the part of `x` here.

```ts
predTrail.every((pred, i) => pred(x[offset + i])) &&
```